crates/fluvio-spu/src/control_plane/message_sink.rs

```diff
@@ -3,26 +3,34 @@ use std::sync::Arc;
 
 use async_lock::Mutex;
 use fluvio_controlplane::sc_api::update_lrs::LrsRequest;
+use fluvio_controlplane::sc_api::update_mirror::MirrorStatRequest;
 
 pub type SharedStatusUpdate = Arc<StatusMessageSink>;
+pub type SharedMirrorStatusUpdate = Arc<StatusMirrorMessageSink>;
 
 /// channel used to send message to sc
 #[derive(Debug)]
-pub struct StatusMessageSink(Mutex<HashSet<LrsRequest>>);
+pub struct MessageSink<R>(Mutex<HashSet<R>>);
```

Comment:
is reason why this is generic instead of enum.  Since this is dynamic, how would receiver would distinguish?

Reply:
Hmm, I think that we could use an enum if we have only one store and use only one function to dispatcher them. 

I added a new store for the mirror status and a new function (`send_mirror_status_back_to_sc`) slightly different from the partition (`LrsRequest`) status (`send_status_back_to_sc`).

what do you think?

Reply:
I think that one good reason to separate it into different stores is to respect the locks by context. wdyt?

Reply:
I renamed `StatusMessageSink` to `StatusLrsMessageSink` to make it easy to understand that this is not a generic store.